### PR TITLE
Add Azure DNS modules

### PR DESCRIPTION
**Pull request by nicholasmhughes**
### What does this PR do?
Add Azure DNS modules for crud operations on DNS zones and record sets.

### What issues does this PR fix or reference?
N/A

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by gtmanfred — APPROVED** [review]
This looks good, can we get some unittests for these modules?

Thanks,
Daniel

**Comment by nicholasmhughes** [comment]
The only unit tests I've done were heavily copy/pasted for (I believe) some `file` module functionality. I don't really know where to start for building Azure module unit tests from scratch. If someone can hold my hand on this, I might try... but I'm not certain if I have the bandwidth to complete them in a timely manner.

**Comment by gtmanfred** [comment]
Look at the Boto modules unit tests.

And I am available (not today,at a conference today) but to help in the
#develop channel on the community slack.  Or in just pms there.

Thanks!
Daniel

**Comment by rallytime** [comment]
@nicholasmhughes Just a bump here. Were you able to get some tests going for this?

**Comment by nicholasmhughes** [comment]
@rallytime , nope... I tried to mess around and duplicate the tests from the Azure SDK, but failed. I don't really have time in the near future to make it happen. As far as this PR, it's been field tested and there are already a lot of Azure modules merged that don't have tests. We can either merge this or close it. I'd prefer the latter obviously. :wink:

File: salt/states/azurearm_dns.py [hunk 1]
```diff
@@ -0,0 +1,685 @@
...
+            azurearm_dns.zone_absent:
+                - name: contoso.com
+                - resource_group: my_rg
+                - connection_auth: {{ profile }}
+
+'''
+
+# Python libs
+from __future__ import absolute_import
+import json
```
Comment (cachedout) [inline salt/states/azurearm_dns.py]: Unused import?

File: salt/states/azurearm_dns.py [hunk 2]
```diff
@@ -0,0 +1,685 @@
...
+import logging
+
+# Salt libs
+import salt.ext.six as six
+import salt.utils
```
Comment (cachedout) [inline salt/states/azurearm_dns.py]: Unused import?

**Review by cachedout — CHANGES_REQUESTED** [review]

**Review by cachedout — APPROVED** [review]

**Comment by cachedout** [comment]
Merged! Thanks, @nicholasmhughes. This is great.

**Pull request merged**